styleguide/javascript/README.md

```diff
@@ -0,0 +1,147 @@
+# JavaScript Styleguide
+
+## Tooling
+
+### Avoid jQuery
```

Comment:
I would like us to make more specific plans about how to go about this in a productive way – e.g. Calle and I have kept using jQuery for sprinkles of theme JS because that's the path of least resistance in the project currently.

Also, considering the discussion we had about "the next person to do uploads MUST adapt CarrierWave to support it", it might be good to set expectations more explicitly. Current phrasing both seems laid back about it ("avoid") and very forceful ("Just don't use it") :) I would suggest going in the more laid-back direction – encourage The New Way but don't "force" devs to switch focus to rewrite existing code they happen to touch.

Perhaps before we merge this PR, someone could rewrite one or a few of our current sprinkles of jQuery in The Desired Way? It would let us all look at it and (I suspect) agree we like it better, and it would also be something we could even point to in the styleguide as an example of how to do things in future.

Perhaps I could do this with the theme JS we added if I get some spare time today – but no promises :)

Reply:
jQuery is adopted still, see https://trello.com/b/ytd5jcWY/tech-radar

Reply:
I'd be happy to update the tech radar if the team is on board with this change.

Reply:
Don't use jQuery in new projects. If jQuery is present in whatever project you're working in, then use it.

Reply:
In new projects we shouldn't do much DOM-manipulation at all...

Reply:
Just don't add jQuery to any new projects. You don't need it.

Reply:
Sounds like a good clarification to make. Don't use it in new projects. Fine to keep using in old ones. Might also be nice to encourage writing new code in existing projects this way, just as long as we're clear that there's no requirement to rewrite huge chunks of existing code just because you need to modify it.

I recently rewrote a side project jQuery to vanilla JS and it was mostly fine. Some things got a little worse and there were some initial gotchas (`document.querySelector(".foo").stuff()` is an error if there's no .foo, unlike jQuery) but definitely felt worth it.